extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxHttpExporter.java

```diff
@@ -0,0 +1,165 @@
+                Consumer<Throwable> onError) {
+
+            client.request(HttpMethod.POST, TRACES_PATH)
+                    .onSuccess(new Handler<>() {
+                        @Override
+                        public void handle(HttpClientRequest request) {
+
+                            HttpClientRequest clientRequest = request.response(new Handler<>() {
+                                @Override
+                                public void handle(AsyncResult<HttpClientResponse> callResult) {
+                                    if (callResult.succeeded()) {
+                                        HttpClientResponse clientResponse = callResult.result();
+                                        clientResponse.body(new Handler<>() {
+                                            @Override
+                                            public void handle(AsyncResult<Buffer> bodyResult) {
+                                                if (bodyResult.succeeded()) {
+                                                    onResponse.accept(new Response() {
+                                                        @Override
+                                                        public int statusCode() {
+                                                            return clientResponse.statusCode();
+                                                        }
+
+                                                        @Override
+                                                        public String statusMessage() {
+                                                            return clientResponse.statusMessage();
+                                                        }
+
+                                                        @Override
+                                                        public byte[] responseBody() {
+                                                            return bodyResult.result().getBytes();
+                                                        }
+                                                    });
+                                                } else {
+                                                    onError.accept(bodyResult.cause());
+                                                }
+                                            }
+                                        });
+                                    } else {
+                                        onError.accept(callResult.cause());
+                                    }
+                                }
+                            })
+                                    .putHeader("Content-Type", contentType);
+
+                            ByteArrayOutputStream os; // TODO: we can probably use Vert.x / Netty buffering here
+                            if (compressionEnabled) {
+                                os = new ByteArrayOutputStream(contentLength);
+                                clientRequest.putHeader("Content-Encoding", "gzip");
```

Comment:
this needs to read the `quarkus.otel.exporter.otlp.traces.compression` property

Reply:
The value of that is passed to the ctor of `VertxHttpExporter` by `OtlpRecorder` (same as for the gRPC exporter). 
Furthermore, the tests already cover this

Reply:
ok